lektor/packages.py

```diff
@@ -143,14 +166,31 @@ def install_local_package(package_root, path):
         dirs = os.listdir(tmp)
         if rv != 0 or len(dirs) != 1:
             raise RuntimeError("Failed to create egg info for local package.")
-        requires = os.path.join(tmp, dirs[0], "requires.txt")
+        requires_path = os.path.join(tmp, dirs[0], "requires.txt")
 
-        # We have dependencies, install them!
-        if os.path.isfile(requires):
-            download_and_install_package(package_root, requirements_file=requires)
+        if os.path.isfile(requires_path):
+            # We have dependencies, install them!
+            return requirements_from_requires_file(requires_path)
     finally:
         shutil.rmtree(tmp)
 
+    # no dependencies to install
+    return None
+
+
+def requirements_from_requires_file(requires_path):
```

Comment:
(Is there a good reason not in in-line this function within `requirements_from_unfinished_editable_install_at_path`?)

Reply:
I think the code reads and explains better what it does this way - the function is mostly self contained and doesn't need the context of the other function where it is used to be understood.